core-web/apps/dotcms-ui/src/app/portlets/dot-categories/dot-categories-table/store/dot-categories-table.store.ts

```diff
@@ -0,0 +1,189 @@
+import { DotMessageService } from '@services/dot-message/dot-messages.service';
+import { DotCategory } from '@dotcms/app/shared/models/dot-categories/dot-categories.model';
+import { PaginatorService } from '@dotcms/app/api/services/paginator';
+import { take } from 'rxjs/operators';
+import { DotActionMenuItem } from '@dotcms/app/shared/models/dot-action-menu/dot-action-menu-item.model';
+
+export interface DotCategoriescTableState {
+    categoriesBulkActions: MenuItem[];
+    categoriesActions: DotActionMenuItem[];
+    selectedCategories: DotCategory[];
+    addToBundleIdentifier: string;
+    tableColumns: DataTableColumn[];
+    categories: DotCategory[];
+    paginationPerPage: number;
+    currentPage: number;
+    totalRecords: number;
+}
+
+@Injectable()
+export class DotCategoriesTableStore extends ComponentStore<DotCategoriescTableState> {
+    constructor(
+        private dotMessageService: DotMessageService,
+        public paginatorService: PaginatorService
+    ) {
+        super(null);
+        this.paginatorService.url = 'v1/categories';
+        this.paginatorService
+            .get()
+            .pipe(take(1))
+            .subscribe((items: DotCategory[]) => {
+                this.setState({
+                    categoriesBulkActions: this.getCategoriesBulkActions(),
+                    categoriesActions: this.getCategoriesActions(),
+                    tableColumns: this.getCategoriesColumns(),
+                    addToBundleIdentifier: '',
+                    selectedCategories: [],
+                    categories: items,
+                    currentPage: this.paginatorService.currentPage,
+                    paginationPerPage: this.paginatorService.paginationPerPage,
+                    totalRecords: this.paginatorService.totalRecords
+                });
+            });
+    }
+    readonly vm$ = this.select((state: DotCategoriescTableState) => state);
+
+    /**
+     * A function that updates the state of the store.
+     * @param state DotCategoryListState
+     * @param selectedCategories DotCategory[]
+     */
+    readonly updateSelectedCategories = this.updater<DotCategory[]>(
+        (state: DotCategoriescTableState, selectedCategories: DotCategory[]) => {
+            return {
+                ...state,
+                selectedCategories
+            };
+        }
+    );
+
+    /** A function that updates the state of the store.
+     * @param state DotCategoryListState
+     * @param categories DotCategory[]
+     */
+    readonly setCategories = this.updater<DotCategory[]>(
+        (state: DotCategoriescTableState, categories: DotCategory[]) => {
+            return {
+                ...state,
+                categories
+            };
+        }
+    );
+
+    /**
+     * > This function returns an observable of an array of DotCategory objects
+     * @returns Observable<DotCategory[]>
+     */
+    public getCategories(filter?: string, currentPage?: number): void {
```

Comment:
This doesn't belong to the store, this is a category service. You might need to end up wrapping the paginator service, I remember we talked about this in our call.

The store is for selects, updaters and effects.